src/mlpack/methods/random_forest/extra_trees_main.cpp

```diff
@@ -0,0 +1,263 @@
+    PRINT_DATASET("predictions") + ", one "
+    "could call "
+    "\n\n" +
+     PRINT_CALL("extra_trees", "input_model", "et_model", "test", "test_set",
+               "test_labels", "test_labels", "predictions", "predictions"));
+
+BINDING_SEE_ALSO("@decision_tree", "#decision_tree");
+BINDING_SEE_ALSO("@hoeffding_tree", "#hoeffding_tree");
+BINDING_SEE_ALSO("@softmax_regression", "#softmax_regression");
+BINDING_SEE_ALSO("Extra Trees on Wikipedia",
+                 "https://en.wikipedia.org/wiki/Random_forest#ExtraTrees");
+PARAM_MATRIX_IN("training", "Training dataset.", "t");
+PARAM_UROW_IN("labels", "Labels for training dataset.", "l");
+PARAM_MATRIX_IN("test", "Test dataset to produce predictions for.", "T");
+PARAM_UROW_IN("test_labels", "Test dataset labels, if accuracy calculation is" 
+"desired.", "L");
+PARAM_MATRIX_IN("training", "Training dataset.", "t");
+PARAM_UROW_IN("labels", "Labels for training dataset.", "l");
+PARAM_MATRIX_IN("test", "Test dataset to produce predictions for.", "T");
+PARAM_UROW_IN("test_labels", "Test dataset labels, if accuracy calculation is"
+                             "desired.",
+              "L");
+
+PARAM_FLAG("print_training_accuracy", "If set, then the accuracy of the model"
+"on the training set will be predicted (verbose must also be specified).","a");
+
+PARAM_INT_IN("num_trees", "Number of trees in the extra trees.", "N", 10);
+PARAM_INT_IN("minimum_leaf_size", "Minimum number of points in each leaf"
+"node.","n", 1);
+PARAM_INT_IN("maximum_depth", "Maximum depth of the tree (0 means no limit).",
+             "D", 0);
+PARAM_MATRIX_OUT("probabilities", "Predicted class probabilities for each"
+                                  "point in the test set.",
+                 "P");
+PARAM_UROW_OUT("predictions", "Predicted classes for each point in the test"
+"set.","p");
+
+PARAM_DOUBLE_IN("minimum_gain_split", "Minimum gain needed to make a split"
+"when building a tree.","g", 0);
+PARAM_INT_IN("subspace_dim", "Dimensionality of random subspace to use for"
+"each split.  '0' will autoselect the square root of data dimensionality.",
+"d", 0);
+
+PARAM_INT_IN("seed", "Random seed.  If 0, 'std::time(NULL)' is used.", "s", 0);
+PARAM_FLAG("warm_start", "If true and passed along with `training` and "
+"`input_model` then trains more trees on top of existing model.",
+"w");
+class ExtraTreesModel
+{
+    public:
+    // The model itself.
+    ExtraTrees<> et;
+    ExtraTreesModel()
+    { /* Nothing yo do. */
+    }
+
+    // Serialize the model.
+    template <typename Archive>
+    void serialize(Archive &ar, const uint32_t /* version */)
+    {
+        ar(CEREAL_NVP(et));
+    }
+};
+
+PARAM_MODEL_IN(ExtraTreesModel, "input_model", "Pre-trained extra trees "
+"algorithm to use for classification.","m");
+PARAM_MODEL_OUT(ExtraTreesModel, "output_model", "Model to save trained "
+"extra trees to.", "M");
+static void mlpackMain()
+{
+    // Check for incompatible input parameters.
```

Comment:
A better comment here would point out why the parameters we are checking are incompatible.